### PR TITLE
nrf: Add board definitions for the Seeed XIAO NRF52840 Sense device.

### PR DESCRIPTION
**Pull request by robert-hh**
It keeps compatibility with the XIAO bootloader by:

- using Soft Device 7.3.0
- reserving 48k memory for the bootloader.

So on double reset a drive pops for uploading an uf2 image or a nrfutil pkg file. Instructions to create it from a hex file are included.

The script download_ble_stack.sh has been adapted to get the version 7.3.0 soft device files. It may have to be executed once before building.

The CPU file system is set to 256k and the pin definitions are adapted. Besides that, it has the common functionality. The on-board sensors and additional flash can be supported by Python scripts.

### CODE REVIEW

File: ports/nrf/modules/manifest.py [hunk 1]
```diff
@@ -1,2 +1,3 @@
-module("_mkfs.py", base_path="$(PORT_DIR)/modules/scripts", opt=3)
+# module("_mkfs.py", base_path="$(PORT_DIR)/modules/scripts", opt=3)
+freeze("$(PORT_DIR)/modules/scripts")
```
Comment (dpgeorge) [inline ports/nrf/modules/manifest.py:2]: It's not clear why this change is needed, what was the reason?  I think just revert it.  If a board needs more scripts then it should specify them in its own `manifest.py`.

Comment (robert-hh) [reply ports/nrf/modules/manifest.py:2]: It's more a test thing, so I can add files to be frozen which do not go into the main line.

Comment (robert-hh) [reply ports/nrf/modules/manifest.py:2]: Reverting it just requires to drop that last commit. Another question about generated files: The XIAO board firmware can be loaded as .uf2 file, which is convenient for users. For the Nordic Semi port, usually the .bin and .hex files are created. So I added instructions in deploy.md to create the .uf2 file.  Is is possible to add .uf2 file creation to a single or few boards as well? I did not check whether other boards support .uf2 file loading as well. The Arduino Nano Connect board could be a candidate.

Comment (robert-hh) [reply ports/nrf/modules/manifest.py]: Reverted.

Comment (dpgeorge) [reply ports/nrf/modules/manifest.py]: > is it possible to add .uf2 file creation to a single or few boards as well?

Yes we can do that, like it's done on samd.

File: ports/nrf/boards/SEEED_XIAO/deploy.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Make and install a firmware package for SEEED nrf52840:
+
+1. Install adafruit-nrfutil with `apt-get install adafruit-nrfutil`
+   The official NRF nrfutil suite used in Makefile did not work for me and
+   failed consistently with Python errors. Maybe a wrong Python version,
+   but obviously it misses version checks.
```
Comment (dpgeorge) [inline ports/nrf/boards/SEEED_XIAO/deploy.md:6]: Please reword this to something like "Install the nrfutil suite.  If this does not work then you can try adafruit-nrfutil ..." (better not to use words like "I" and "me" in docs).

File: ports/nrf/boards/SEEED_XIAO/deploy.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   - Upload the firmware through USB serial with:
+
+     `nrfutil dfu serial -pkg firmware.zip --port /dev/ttyACM0 --baudrate 460800 -sb`
+
+In case you lost or have overwritten the XIAO bootloader, you can find it
+at https://github.com/Seeed-Studio/Adafruit_nRF52_Arduino.git in different
+formats. Using a JLINK adapter or interface, you can upload the hex version.
+The bootloader is as well available through the Arduino IDE.
+
+# Create a uploadable uf2 file.
```
Comment (dpgeorge) [inline ports/nrf/boards/SEEED_XIAO/deploy.md:32]: I'm not sure why this section is needed, since you added code in mpconfigboard.mk to automatically build the uf2.

File: ports/nrf/boards/SEEED_XIAO/mpconfigboard.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+#define MICROPY_HW_PWM1_NAME        "PWM1"
+#define MICROPY_HW_PWM2_NAME        "PWM2"
+
+#define HELP_TEXT_BOARD_LED         "1,2,3,4"
+
+// Enabling these VID/PID values require to include mpconfigboard.h into
+// usb_descriptors.c. But there is an open discussion on whether
+// specific VID/PID values are wanted.
+// 1200BPS touch is not supported at the moment
+// in the USB driver, but at least the flasg is set.
```
Comment (dpgeorge) [inline ports/nrf/boards/SEEED_XIAO/mpconfigboard.h:77]: flasg -> flag

**Comment by jimmo** [comment]
> The name of this board will clash with the samd board of the same name, so it should be changed, at the very least so users do not get confused.

Created https://github.com/micropython/micropython/pull/10213 to rename the SAMD board and https://github.com/micropython/micropython/pull/10213 to update the images (including adding the image for this nRF board).

@robert-hh Can you please rename the board added in this PR to `seeed_xiao_nrf52`. Thanks!

File: ports/nrf/modules/manifest.py (hunk 1 above)
Comment (robert-hh) [reply ports/nrf/modules/manifest.py]: At the SAMD port the .uf2 file is created for all boards. But here it would only be needed for the SEEED_XIAO_NRF52 board.

**Comment by robert-hh** [comment]
I renamed the nrf board to SEEED_XIAO_NRF52.

File: ports/nrf/boards/SEEED_XIAO/deploy.md (hunk 3 above)
Comment (robert-hh) [reply ports/nrf/boards/SEEED_XIAO/deploy.md:32]: I'll remove it, but keep the note about reinstalling the bootloader.

**Comment by dpgeorge** [comment]
Thanks for updating.
